android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt

```diff
@@ -286,6 +288,23 @@ class AddonsManagerAdapter(
             }
             holder.statusBlocklistedView.isVisible = true
         }
+
+        if (addon.isDisabledAsNotCorrectlySigned()) {
+            holder.statusUnsignedView.findViewById<TextView>(R.id.add_on_status_unsigned_message).text =
+                context.getString(
+                    R.string.mozac_feature_addons_status_unsigned,
+                    addonName,
+                )
+            holder.statusUnsignedView.findViewById<TextView>(
+                R.id.add_on_status_unsigned_learn_more_link,
+            ).setOnClickListener {
+                addonsManagerDelegate.onLearnMoreLinkClicked(
+                    AddonsManagerAdapterDelegate.LearnMoreLinks.ADDON_NOT_CORRECTLY_SIGNED,
+                    addon,
+                )
+            }
+            holder.statusUnsignedView.isVisible = true
+        }
```

Comment:
Same as https://github.com/mozilla-mobile/firefox-android/pull/3840/files#r1343895119, looks like the only changes that we have here are the error message and the link, could we reuse the previous UI block?

Reply:
Maybe ?

``` kotlin
        val (disabledErrorStringId, disabledStatus) = if (addon.isDisabledAsBlocklisted()) {
            R.string.mozac_feature_addons_status_blocklisted to AddonsManagerAdapterDelegate.LearnMoreLinks.BLOCKLISTED_ADDON
        } else if (addon.isDisabledAsNotCorrectlySigned()) {
            R.string.mozac_feature_addons_status_unsigned to AddonsManagerAdapterDelegate.LearnMoreLinks.ADDON_NOT_CORRECTLY_SIGNED
        } else null to null

        if (disabledErrorStringId != null && disabledStatus != null) {
            holder.statusUnsignedView.findViewById<TextView>(R.id.add_on_disabled_status_message).text =
                context.getString(
                    disabledErrorStringId,
                    addonName,
                )
            holder.statusUnsignedView.findViewById<TextView>(
                R.id.add_on_disabled_status_learn_more_link,
            ).setOnClickListener {
                addonsManagerDelegate.onLearnMoreLinkClicked(
                    disabledStatus,
                    addon,
                )
            }
            holder.statusUnsignedView.isVisible = true
        }
```